File: app/src/main/java/com/example/leeeyou/zhihuribao/data/service/ZhiHuRiBaoService.java
package com.example.leeeyou.zhihuribao.data.service;

import com.example.leeeyou.zhihuribao.data.model.RiBao;
import com.example.leeeyou.zhihuribao.data.model.StoryDetail;

import retrofit.http.GET;
import retrofit.http.Path;
import rx.Observable;

/**
 * Created by leeeyou on 16/5/10.
 * <p>
 * 知乎日报业务服务类
 */
public interface ZhiHuRiBaoService {

    @GET("/api/4/news/before/{dateTime}")
    Observable<RiBao> getLatestRiBao(@Path("dateTime") String dateTime);

    @GET("/api/4/news/{id}")
    Observable<StoryDetail> getStoryDetailById(@Path("id") int id);

}

File: app/src/main/java/com/xyz/leeeyou/zhihuribao/data/service/ZhiHuRiBaoService.kt
package com.xyz.leeeyou.zhihuribao.data.service

import com.xyz.leeeyou.zhihuribao.data.model.ribao.ZhiHuDaily
import com.xyz.leeeyou.zhihuribao.data.model.ribao.ZhiHuDailyDetail
import retrofit.http.GET
import retrofit.http.Path
import rx.Observable

/**
 * ClassName: ZhiHuRiBaoService
 * Description: [ ZhiHuRiBao ] module service
 *
 * Author:      leeeyou
 * Date:        2018/4/24 15:14
 */
interface ZhiHuRiBaoService {
    @GET("/api/4/news/before/{dateTime}")
    fun getLatestZhiHuDaily(@Path("dateTime") dateTime: String): Observable<ZhiHuDaily>

    @GET("/api/4/news/{id}")
    fun getZhiHuDailyDetailById(@Path("id") id: Int): Observable<ZhiHuDailyDetail>
}